### DISCUSSION TITLE
Windows ARM32 Release build?

### DISCUSSION THREAD

**Discussion by BaRRaKudaRain** [86Box/86Box#3471] [General]
Any luck of getting prebuilt Windows ARM32 build of 86box?

**Comment by dharmOS** [comment]
I would prefer a Windows ARM64 compile if possible, analogous to the exisiting ARM64 build for Linux AppImage.

**Comment by MagnetarRocket** [comment]
I'd say both a 32 & 64 bit build on ARM, should be easy if the compiler doesn't spit out errors.

**Comment by thorr2** [comment]
Just got a Surface Pro 11.  I am hoping that now that Microsoft is embracing the Snapdragon Elite Arm architecture that we can get a Windows Arm64 compiled version moving forward.  I could maybe compile it myself, but I would need to learn how, and that would not benefit anyone else.

**Reply by dxgldotorg** [reply]
Just as a heads-up, the latest versions of Windows 11 do not support Arm32 processes anymore; the support was entirely ripped out not just the Windows Store app support.

**Comment by waltje** [comment]
There is nothing magical about compiling for ARM and ARM64, I have been doing it for years.
For ARM(32) on Windows, there's a catch, however, because Microsoft tried to discourage developersfrom builting applications (native apps, not CLR apps) for that platform. Those 'evils' can be undone,and then you can just compile for Arm32 (i.e. windows 8.1) without issues.
I am running stuff on my cute lil Surface RT's without problems! :)
--waltje

    On Friday, June 28, 2024 at 05:22:05 PM EDT, thorr2 ***@***.***> wrote:  
 
 


Just got a Surface Pro 11. I am hoping that now that Microsoft is embracing the Snapdragon Elite Arm architecture that we can get a Windows Arm64 compiled version moving forward. I could maybe compile it myself, but I would need to learn how, and that would not benefit anyone else.

**Reply by thorr2** [reply]
Thanks!  I would love to learn how to do this so I can get all my apps working better.  Hopefully I can find a good youtube video or something that explains the process from start to finish.  I compile all the time in Linux with gcc, but I haven't touched this in Windows and not sure what I need and how to do it and make it work with graphics, etc.  Maybe it is super simple like Linux sometimes is, but I don't know.  Perhaps I can just ask Co-pilot and my computer will tell me what to do, lol.

**Comment by cold-brewed** [comment]
We don't currently provide any builds for Windows ARM because we don't have any means for automated building.

However, you can build them yourself for ARM64 by following the [build guide](https://86box.readthedocs.io/en/latest/dev/buildguide.html). One notable deviation for ARM64 windows: Instead of using the `MINGW64` environment of msys2, you'll need to use `CLANGARM64` instead. You'll also need to specify `-D NEW_DYNAREC=ON` as only the new dynarec will work on arm systems. Otherwise the rest of the build guide applies the same as building for x86-64 windows (such as the need to install the required dependencies).

This is fairly recent so the build guide hasn't been updated yet for windows on ARM64 but the build system itself supports it.

**Reply by thorr2** [reply]
Thank you very much!

**Comment by talynone** [comment]
Now that GitHub provides free public Windows ARM runners, is it possible to get an automated ARM64 build?

**Reply by jriwanek** [reply]
You mean like ` 86Box-NDR-Dev-Windows-arm64-gha6633`?
Because we've offered that for ages.

**Reply by talynone** [reply]
> You mean like ` 86Box-NDR-Dev-Windows-arm64-gha6633`? Because we've offered that for ages.

Oh nice! I was only checking releases, should have checked the nightlies, thanks!

**Comment by waltje** [comment]
I once created a Windows ARM32 (Windows RT, aka Windows 8.1 for the Surface RT) for myemulator (VARCem) which worked fine.  Should be easy for 86Box as well.
The problem with this is, that Microsoft (deliberately) crippled the build system for this variantof Windows, to 'force' devs to develop Metro apps (.Net) for it instead of native ones.  So, youhad to 'work around' this before you could do an actual build.
Other than, it works just fine.. "its just Windows" ;-)
--Fred
    On Friday, December 26, 2025 at 01:34:52 PM EST, dxgldotorg ***@***.***> wrote:  
 
 


Just as a heads-up, the latest versions of Windows 11 do not support Arm32 processes anymore; the support was entirely ripped out not just the Windows Store app support.
